src/tools/pkcs11-tool.c

```diff
@@ -4166,6 +4182,99 @@ parse_ec_pkey(EVP_PKEY *pkey, int private, struct gostkey_info *gost)
 
 	return 0;
 }
+
+#ifdef ENABLE_OPENSSL
+/* return PKCS11 key type based on OpenSSL EVP_PKEY type
+ * which are support by PKCS11 and OpenSSL used when compiling
+ * PKCS11 returns CKK_EC_EDWARDS for both ED25529 and X448
+ * and CKK_EC_MONTGOMERY for X25519 and X448, use pk_type.
+ */
+static CK_RV
+evp_pkey2ck_key_type(EVP_PKEY *pkey, CK_KEY_TYPE *type, int *pk_type)
+{
+	if (!pkey || !pk_type || !type)
+		return CKR_GENERAL_ERROR;
+
+	*pk_type = EVP_PKEY_base_id(pkey);
+
+#if defined(EVP_PKEY_RSA) && defined(CKK_RSA)
```

Comment:
do we need to check for the CKK_ being defined when we define them ourselves in pkcs11.h?

Reply:
As suggested, dropping "defined(CKK_*)" tests as OpenSC defines these.